.yamato/package-test.yml

```diff
@@ -15,8 +18,15 @@ test_{{ platform.name }}_{{ editor.version }}:
     UPMCI_ENABLE_APV_CLEAN_CONSOLE_TEST: 1
 {% endif %}
   commands:
-     - npm install upm-ci-utils@stable -g --registry https://artifactory.prd.cds.internal.unity3d.com/artifactory/api/npm/upm-npm
-     - upm-ci package test --unity-version {{ editor.version }} --package-path com.unity.formats.alembic
+    - npm install upm-ci-utils@stable -g --registry https://artifactory.prd.cds.internal.unity3d.com/artifactory/api/npm/upm-npm
+# 2019.4 and 2020.3 doesn't have Silicon version Editor, so download x64 version for them even on Silicon Mac. On other platforms, always download x64 version.
+{% if platform.model != "M1" or editor.version == "2019.4" or editor.version == "2020.3" %}
+    - upm-ci package test --unity-version {{ editor.version }} --package-path com.unity.formats.alembic
+{% else %}
+# Explicitly download Silicon version Editor for 2021.3+(included) on Silicon Mac
+    - unity-downloader-cli -u {{ editor.version }} -a arm64 -c Editor --fast --wait
```

Comment:
`upm-ci` call `unity-downloader-cli` under the hood to download Editor before running tests. In theory `unity-downloader-cli` can detect what the current platform is such as arm64 Mac or x64 Mac and then decide which Editor to download.
But currently `unity-downloader-cli` has a bug and it always download `x64` version Editor even on Silicon Mac. This is not what we want because we want to test Silicon Editor on Silicon Mac. 
That's why I use a workaround here which is when on Silicon Mac `platform.model == M1 and editor.version != "2019.4" and editor.version != "2020.3"` (see explanation I made for Lucille's comment) , `unity-downloader-cli` explicitly download arm64 version Editor for upm-ci to use for testing.

Reply:
I just put more comments in the codes, hope it will be more clear.